compiler/qsc_qasm3/src/semantic/symbols.rs

```diff
@@ -89,15 +93,73 @@ impl std::fmt::Display for SymbolId {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)]
 pub struct Symbol {
     pub name: String,
     pub span: Span,
     pub ty: Type,
     pub qsharp_ty: crate::types::Type,
     pub io_kind: IOKind,
+    // Used for const evaluation.
+    // This field should only be Some(_)
+    // if the symbol is const.
+    value: Option<Rc<Expr>>,
 }
```

Comment:
Does it make sense for the table itself to hold the values rather than the symbol itself?